src/snowflake/cli/plugins/nativeapp/codegen/snowpark/callback_source.py.jinja

```diff
@@ -0,0 +1,117 @@
+import functools
+import sys
+from typing import Any, Callable, List
+
+try:
+    import snowflake.snowpark
+except ModuleNotFoundError as exc:
+    print(
+        "An exception occurred while importing snowflake-snowpark-python package: ",
+        exc,
+        file=sys.stderr,
+    )
+    sys.exit(1)
+
+found_correct_version = hasattr(
+    snowflake.snowpark.context, "_is_execution_environment_sandboxed_for_client"
+) and hasattr(snowflake.snowpark.context, "_should_continue_registration")
+
+if not found_correct_version:
+    print(
+        "Did not find the minimum required version for snowflake-snowpark-python package. Please upgrade to v1.15.0 or higher.",
+        file=sys.stderr,
+    )
+    sys.exit(1)
+
+orig_globals = globals().copy()
```

Comment:
We need to snapshot before we pollute the global space with `found_correct_version`